Sources/MeiliSearch/Model/SearchRules.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+
+public struct SearchRules: Codable, Equatable {
+  var index: String = "*"
+  var filter: String?
+
+  init(_ index: String, filter: String? = nil) {
```

Comment:
I thought a lot about this, and I think the version without the name would be more `easy` to read.

```swift
SearchRules("movies")
SearchRules(index: "movies")
```

because we can compose these rules with each other:

```swift
SearchRulesGroup([
  SearchRules("movies", filter: "genres = Drama AND id = 1"),
  SearchRules("other_index", filter: "type = anything")
])

// vs.

SearchRulesGroup([
  SearchRules(index: "movies", filter: "genres = Drama AND id = 1"),
  SearchRules(index: "other_index", filter: "type = anything")
])
```